### ISSUE TITLE
pkg.installed fails when unholding and test=true

### ISSUE THREAD

**Issue by Sakorah** [saltstack/salt#41651]
Hi,

I use pkg.installed to manage packages and hold their version. The installed version is determined from a pillar. I use the following states to accomplish that:
```
{% if salt['pillar.get']('elkstack:ls-version') != salt['pkg.version']('logstash') and salt['pkg.version']('logstash') != '' %}
logstash-unhold:
  pkg.installed:
    - name: logstash
    - hold: False
    - require:
      - pkgrepo: elkstack-repos
{% endif %}

logstash-install:
  pkg.installed:
    - name: logstash
    - version: {{salt['pillar.get']('elkstack:ls-version')}}
    - listen_in: 
      - service: logstash-service
    - require:
      - pkgrepo: elkstack-repos
{% if salt['pillar.get']('elkstack:ls-version') != salt['pkg.version']('logstash') and salt['pkg.version']('logstash') != '' %}
      - pkg: logstash-unhold
{% endif %}

logstash-hold:
  apt:
    - name: logstash
    - held
    - onchanges:
      - pkg: logstash-install
```
First state checks if installed version differs from the pillar and if so unholds the package. Second state installs the package and the third state holds the package again.

When I try to do a dry run with `salt '*' state.apply test=true`I get the following error:
```
          ID: logstash-unhold
    Function: pkg.installed
        Name: logstash
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1746, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1704, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/dist-packages/salt/states/pkg.py", line 1326, in installed
                  if hold_ret[x]['changes']]
              TypeError: string indices must be integers, not str
     Started: 16:39:23.464969
    Duration: 27.297 ms
     Changes:
```
Tho it seems that without `test=true` everything works but its a bit a pain when the dry run for a sw update does fail this hard so all other depending state fail as well...

### Versions Report
Saltmaster and minion are on the same version on the same machine
```
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.6 (default, Oct 26 2016, 20:30:19)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 4.4.0-79-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by Ch3LL** [comment]
@Sakorah i'm having a hard tiem replicating this using the following state:

```
salt-minion-unhold:
  pkg.installed:
    - name: salt-minion
    - hold: False

salt-minion-install:
  pkg.installed:
    - name: salt-minion
    - version: '2016.11.5'

salt-minion-hold:
  apt:
    - name: salt-minion
    - held
    - onchanges:
      - pkg: salt-minion-install
```

And salt version 2016.11.4 is installed. Is my test case wrong as you can see?

**Comment by Ch3LL** [comment]
@garethgreenaway pinged me and stated he can replicate this behavior so he will look at a fix for this.

**Comment by garethgreenaway** [comment]
This issue should be considered resolved and the fix is available on 2016.11. If the issue is not considered resolved or persists, please feel free to re-open this issue.
